### PR TITLE
test: add a test verifying that the tests are run in the checked mode

### PR DESCRIPTION
**Pull request by vsavkin**

### CODE REVIEW

File: modules/angular2/test/dev_mode_spec.ts [hunk 1]
```diff
@@ -11,10 +11,23 @@ import {
...
 export function main() {
   describe('dev mode', () => {
     it('is enabled in our tests by default', () => { expect(assertionsEnabled()).toBe(true); });
   });
+
+  if (IS_DART) {
+    describe('checked mode', () => {
+      it('is enabled in our tests', () => {
+        try {
+          var s: string = <any>42;
```
Comment (jeffbcross) [inline modules/angular2/test/dev_mode_spec.ts]: Won't Dart Analyzer complain that `s` is not used?

Comment (jeffbcross) [inline modules/angular2/test/dev_mode_spec.ts]: Oh, nm, I see that you pushed a correction.

**Comment by IgorMinar** [comment]
lgtm

**Comment by jelbourn** [comment]
Closed via 442d6866da14749588b8021259a6c7c8125ef782

**Pull request closed without merging**